client/my-sites/controller.js

```diff
@@ -249,7 +228,23 @@ module.exports = {
 			basePath = route.sectionify( context.pathname );
 		}
 
-		renderNavigation( context, basePath, basePath );
+		context.secondary = React.createElement( ReduxProvider, { store: context.store },
+			React.createElement( NavigationComponent, {
+				layoutFocus,
+				path: context.path,
+				allSitesPath: basePath,
+				siteBasePath: basePath,
+				user,
+				sites
+			} )
+		);
+
+		// Render the My Sites navigation in #secondary
+		ReactDom.render(
+			context.secondary,
+			document.getElementById( 'secondary' )
+		);
+
```

Comment:
My approach would be to provide a `makeNavigation` middleware to be invoked explicitly by isomorphic sections that populates `context.secondary` (that's #3550) while keeping `navigation` for non-iso ones. I think we should then delete `context.primary` and `secondary` at `renderSeparateTrees` stage (after rendering) so we don't run into trouble once other section groups (e.g. reader, me) want to populate `context.secondary` with their sidebars.